Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -271,7 +271,36 @@ final class DeclarationTests: ParserTestCase {
         struct A {
           nonisolated(unsafe) let b = 0
           nonisolated(unsafe) var c: Int { 0 }
-          nonisolated(unsafe) let d = 0
+          nonisolated(<#identifier#>) let d = 0
+        }
+        """
+    )
+  }
+
+  func testNonisolatedNonSendingParsing() {
+    assertParse(
+      """
+      nonisolated(nonsending) let a = 0
+
+      struct A {
+        nonisolated(nonsending) let b = 0
+        nonisolated(nonsending) var c: Int { 0 }
+        nonisolated(1️⃣sending) let d = 0
+      }
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "expected identifier in modifier",
+          fixIts: ["replace 'sending' with identifier"]
```

Comment:
Hmm, this error message isn’t great. I would prefer if we just accept any identifier here in the parser and let ASTGen diagnose unknown identifiers as the detail.

Reply:
Yeah, that's why I was asking how to make it so it lists both `unsafe` and `nonsending` now, `self.expect` accepts multiple choices but only one "default".

Reply:
Something to consider here before we make a final decision: if we did allow any and all identifiers here, it would mean that we'd always parse nonisolated as a declaration modifier, which is a break from what C++ parser is doing and is not always correct. 

Consider this situation:

```
func nonisolated() {
}

let v: Int = 42

nonisolated(v)
func test() -> Int { 42 }
```

This is currently a call + a decl but it would be interpreted as a a decl with a modifier with this change.

Reply:
OK, let’s just stick with the current solution. It’s not the best but maybe good enough for now. And I don’t want to block this PR on improving this.

Reply:
Sounds good!